src/Session.hs

```diff
@@ -92,10 +92,10 @@ isSuccess :: Maybe Summary -> Bool
 isSuccess = not . isFailure
 
 parseSummary :: String -> Maybe Summary
-parseSummary r = case reverse $ lines r of
-  x : _ | Just summary <- readMaybe (dropAnsiEscapeSequences x) -> Just summary
-  _ -> Nothing
+parseSummary = findJust . (map $ readMaybe . dropAnsiEscapeSequences) . take 3 . reverse . lines
```

Comment:
What about skipping the `take 3`.  Would it make it even more robust?  Is there anything that can go wrong?